sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/state/SymmetricHashJoinStateManager.scala

```diff
@@ -443,6 +444,7 @@ class SymmetricHashJoinStateManager(
 
   /** Helper trait for invoking common functionalities of a state store. */
   private abstract class StateStoreHandler(stateStoreType: StateStoreType) extends Logging {
+    private var stateStoreProvider: StateStoreProvider = _
```

Comment:
Hmm, is this change related to state data source reader?

Reply:
Oh, nvm, I think I know where this is used now.

Reply:
Yeah that is actually tricky but there does not seem to be any alternative.